File: examples/cpp/my-app/src/models/Comment.cpp
#include "models/Comment.h"
#include "utils/DateUtils.h"

namespace models {

Comment::Comment(int id, int taskId, int authorId, const std::string& content)
    : id(id), taskId(taskId), authorId(authorId), content(content),
      createdAt(utils::DateUtils::now()) {}

int Comment::getId() const {
    return id;
}

int Comment::getTaskId() const {
    return taskId;
}

int Comment::getAuthorId() const {
    return authorId;
}

std::string Comment::getContent() const {
    return content;
}

std::time_t Comment::getCreatedAt() const {
    return createdAt;
}

void Comment::setContent(const std::string& newContent) {
    content = newContent;
}

} // namespace models

File: examples/python/my-app/src/models/comment.py
"""Comment model."""

from dataclasses import dataclass
from datetime import datetime
from typing import Optional

@dataclass
class Comment:
    """Comment entity for tasks."""
    
    id: int
    task_id: int
    author_id: int
    content: str
    created_at: datetime
    updated_at: Optional[datetime] = None
    is_edited: bool = False
    
    def edit(self, new_content: str) -> None:
        """Edit the comment content."""
        self.content = new_content
        self.is_edited = True
        self.updated_at = datetime.now()
    
    def get_preview(self, max_length: int = 100) -> str:
        """Get a preview of the comment."""
        if len(self.content) <= max_length:
            return self.content
        return self.content[:max_length] + "..."
